### ISSUE TITLE
Search refinement for author links

### ISSUE THREAD

**Issue by tclayton33** [emory-libraries/blacklight-catalog#1400]
As a user, when I click on an author's name in the search results or on an item record, I want all works by that author to show up (including when they are tagged as editor, contributor, additional author, etc.), so I can see everything that author has written and contributed to.

Currently, clicking on an author link in full record display may produce different results depending on how the name is recorded in the bibliographic record and if the name appears as primary author/creator or additional author/creator. 

Example - compare the links that show up after an author search for [Cronan, Todd](https://search.libraries.emory.edu/?utf8=%E2%9C%93&search_field=author&q=cronan%2C+todd):
Author/Creator: **[Cronan, Todd, author](https://search.libraries.emory.edu/?f%5Bauthor_display_ssim%5D%5B%5D=Cronan%2C+Todd%2C+author)**  (3 results)
Additional Author/Creator: **[Cronan, Todd, degree supervisor](https://search.libraries.emory.edu/?f%5Bauthor_addl_ssim%5D%5B%5D=Cronan%2C+Todd)** (4 results)
Additional Author/Creator: **[Cronan, Todd, contributor](https://search.libraries.emory.edu/?f%5Bauthor_addl_ssim%5D%5B%5D=Cronan%2C+Todd)** (4 results)

The marc subfields that need to be included in an exact search string for 100s, 700s are below. Everything else can be excluded.
ǂa Personal name
ǂb Numeration
ǂc Titles and other words associated with a name
ǂd Dates associated with a name

This ticket combines enhancement requests [35](https://app.zenhub.com/workspaces/blacklight-catalog-5f5f84a8a6d29939a0bc5d78/issues/gh/emory-libraries/librarysearch-enhance/35) and [28](https://app.zenhub.com/workspaces/blacklight-catalog-5f5f84a8a6d29939a0bc5d78/issues/gh/emory-libraries/librarysearch-enhance/28)

Additional names for testing and acceptance criteria - clicking on these author's names within a record should produce the same number of results.

- [ ] Kim, Grace Ji-Sun 1969
- [ ] Allan Boesak
- [ ] Carol Newsom
- [ ] Martin, Anthony J., 1960-

**Comment by abelemlih** [comment]
@tclayton33 @rotated8 I researched a solution for this ticket, and noticed this field we currently index in the catalog:

```
to_field 'author_ssim', extract_marc("100abcdq:110abd:111acd:700abcdq:710abd:711acd"), trim_punctuation
```

for each document, it concatenates the main author as well as additional authors, without including the `$e - Relator term (R)` subfield. If we were to switch the author/additional author links to perform a search using `author_ssim` across the repository for that author's name, it would return all documents where that author is either the main author or a contributor, and the results will be the same.
 
Would this solution cover all cases? Do you forsee any issues using `author_ssim`? For the Cronan, Todd example above, this is what a solution would look like: [https://search.libraries.emory.edu/?f[author_ssim][]=Cronan%2C+Todd](https://search.libraries.emory.edu/?f[author_ssim][]=Cronan%2C+Todd)

Thanks!

**Comment by tclayton33** [comment]
@abelemlih I think this solution should work. Author_ssim is the field we are using for the links in the author/creator facet so I would expect the same behavior with respect to number of results, correct? I think one use case that wouldn't work as well as people are hoping is for Kim, Grace Ji-Sun. But that seems to be an issue with our underlying metadata - our name headings for her aren't consistent, which shows up in the facets: https://search.libraries.emory.edu/?utf8=%E2%9C%93&search_field=author&q=+Kim%2C+Grace+Ji-Sun

I'd want to confirm with testing on blackcat-test, but I believe this solution does solve the underlying complaints related to $e being problematic and that the author links aren't including both main author and additional authors.

**Comment by abelemlih** [comment]
@tclayton33 yes, `author_ssim` is the field we are using for the links in the author/creator facet, so we should expect similar results. I will move forward with implementation and let you know once a release is ready for deployment

**Comment by tclayton33** [comment]
per conversation with Ayoub, have modified the work estimate from 8 to 13

**Comment by tclayton33** [comment]
4/2/25 moving this from ready to backlog (until we are ready to resume work on this ticket)
